src/main/java/com/google/cloud/genomics/dataflow/readers/bam/Reader.java

```diff
@@ -75,11 +91,11 @@ public void process() throws IOException {
 
   void openFile() throws IOException {
     LOG.info("Processing shard " + shard);
-    final SamReader reader = BAMIO.openBAM(storageClient, shard. file, stringency);
+    final SamReader reader = BAMIO.openBAM(storageClient, shard. file, 
+        options.getStringency());
     iterator = null;
-    procesingUnmapped = shard.contig.referenceName.equals("*");
     if (reader.hasIndex() && reader.indexing() != null) {
-      if (procesingUnmapped) {
+      if (filter == Filter.UNMAPPED_ONLY) {
```

Comment:
In this case unmapped only - so we ONLY query for unmapped reads here. The other branch is for reading normal reads (and if options tells then also mixing in the unmapped ones)